modules/rntuple.mjs

```diff
@@ -92,55 +99,241 @@ class RBufferReader {
+  this.description = reader.readString();
+  this.writer = reader.readString();
+  // TODO: Remove debug logs before finalizing
+  console.log('Name:', this.name);
+  console.log('Description:', this.description);
+  console.log('Writer:', this.writer);
+}
+_readFieldDescriptors(reader) {
+  this.fieldListSize = reader.readS64(); // signed 64-bit
+  const fieldListIsList = this.fieldListSize < 0;
+
+  if (!fieldListIsList)
+    throw new Error('Field list frame is not a list frame, which is required.');
+
+  const fieldListCount = reader.readU32(); // number of field entries
+  console.log('Field List Count:', fieldListCount);
+
+  // List frame: list of field record frames
+
+  const fieldDescriptors = [];
+  for (let i = 0; i < fieldListCount; ++i) {
+    const fieldVersion = reader.readU32(),
+    typeVersion = reader.readU32(),
+    parentFieldId = reader.readU32(),
+    structRole = reader.readU16(),
+    flags = reader.readU16(),
+
+    fieldName = reader.readString(),
+    typeName = reader.readString(),
+    typeAlias = reader.readString(),
+    description = reader.readString();
+    let arraySize = null, sourceFieldId = null, checksum = null;
+
+    if (flags & 0x1) arraySize = reader.readU32();
+    if (flags & 0x2) sourceFieldId = reader.readU32();
+    if (flags & 0x4) checksum = reader.readU32();
+
+     fieldDescriptors.push({
+        fieldVersion,
+        typeVersion,
+        parentFieldId,
+        structRole,
+        flags,
+        fieldName,
+        typeName,
+        typeAlias,
+        description,
+        arraySize,
+        sourceFieldId,
+        checksum
+    });
+    console.log(`Field ${i + 1}:`, fieldName, '&&', typeName);
+}
+  this.fieldDescriptors = fieldDescriptors;
+}
+_readColumnDescriptors(reader) {
+  this.columnListSize = reader.readS64(); // signed 64-bit
+  const columnListIsList = this.columnListSize < 0;
+  if (!columnListIsList)
+    throw new Error('Column list frame is not a list frame, which is required.');
+  const columnListCount = reader.readU32(); // number of column entries
+  console.log('Column List Count:', columnListCount);
+  const columnDescriptors = [];
+  for (let i = 0; i < columnListCount; ++i) {
+  const coltype = reader.readU16(),
+  bitsOnStrorage = reader.readU16(),
+  fieldId = reader.readU32(),
+  flags = reader.readU16(),
+  representationIndex = reader.readU16();
+
+   let firstElementIndex = null, minValue = null, maxValue = null;
+  if (flags & 0x1) firstElementIndex = reader.readU64();
+  if (flags & 0x2){
+    minValue = reader.readF64();
+    maxValue = reader.readF64();    
+  }
+
+  columnDescriptors.push({
+    coltype,
+    bitsOnStrorage,
+    fieldId,
+    flags,
+    representationIndex,
+    firstElementIndex,
+    minValue,
+    maxValue,
+    isDeferred: (flags & 0x01) !== 0,
+    isSuppressed: (firstElementIndex !== null && firstElementIndex < 0)
+  }); 
+ }
+ this.columnDescriptors = columnDescriptors;
+}
+_readAliasColumn(reader){
```

Comment:
I would add the reading of alias columns, extra type info and cluster groups in a future PR, not this one.
This one already has enough changes with the fields and the columns.

Also, please add some tests regarding fields and columns in `rntuple_test.js` to be sure you are deserializing them correctly. (you are free to split fields and columns to 2 separate PRs as well, if you prefer).